lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -4193,6 +4195,26 @@ ParseResult FIRCircuitParser::parseParameterList(ArrayAttr &resultParameters) {
   return success();
 }
 
+/// class ::= 'class' id ':' info? INDENT portlist simple_stmt_block DEDENT
+ParseResult FIRCircuitParser::parseClass(CircuitOp circuit, unsigned indent) {
+  StringAttr name;
+  SmallVector<PortInfo, 8> portList;
+  SmallVector<SMLoc> portLocs;
+  LocWithInfo info(getToken().getLoc(), this);
+  consumeToken(FIRToken::kw_class);
+  if (parseId(name, "expected class name") ||
+      parseToken(FIRToken::colon, "expected ':' in class definition") ||
+      info.parseOptionalInfo() || parsePortList(portList, portLocs, indent))
+    return failure();
+  auto builder = circuit.getBodyBuilder();
+  auto classOp = builder.create<ClassOp>(info.getLoc(), name, portList);
+  deferredModules.emplace_back(
+      DeferredModuleToParse{classOp, portLocs, getLexer().getCursor(), indent});
+  if (skipToModuleEnd(indent))
+    return failure();
+  return success();
```

Comment:
Only if you prefer, consider:
```suggestion
  return success(skipToModuleEnd(indent));
```